### ISSUE TITLE
TvZ and TvT Crash

### ISSUE THREAD

**Issue by Eliyahu95** [jncraton/BWMetaAI#56]
Everytime I play as Terran with Bio (Marine Medic etc.) against the Zerg AI, the game is collapsing. When I use Mech (Goliaths, Tanks etc.), the game is running correct.

TvT - Game crash as soon as the Terran AI retaliates back.

Game Version 1.23.10.12409, Remastered.

**Comment by jncraton** [comment]
Thanks for sharing this. I haven't been able to replicate this exactly, but I did see at least one stall when playing a bio TvZ. I'll see if I can isolate the issue. We did make some adjustments to Zerg research and upgrades recently that could be related. You might try [2.4](https://github.com/jncraton/BWMetaAI/releases/tag/v2.4) for a more reliable experience.

There is a known (internal to BW) issue that can cause freezes when playing against Terran. If an SCV is destroyed while building a building, the building will never be completed and other progress will be impeded. This can very easily happen in TvT with Vulture run-bys.
